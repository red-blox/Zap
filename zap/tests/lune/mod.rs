use std::fs;

use insta::{assert_debug_snapshot, glob, Settings};
use lune::Runtime;

pub fn run_lune_test(input: &str, no_warnings: bool, insta_settings: Settings) {
	let script = zap::run(input, no_warnings);

	assert!(script.code.is_some(), "No code generated!");

	let code = script.code.as_ref().unwrap();
	let mut runtime = Runtime::new().with_args(vec![&code.server.code, &code.client.code]);

	let result = tokio::runtime::Runtime::new()
		.expect("Unable to setup tokio")
		.block_on(async { runtime.run("zap", include_str!("./base.luau")).await })
		.map(|value| value.0);

	insta_settings.bind(|| assert_debug_snapshot!(result))
}

#[test]
fn test_compile() {
	glob!(env!("CARGO_MANIFEST_DIR"), "tests/files/*.zap", |path| {
		let input = fs::read_to_string(path).unwrap();

		let mut insta_settings = Settings::new();
		insta_settings.set_prepend_module_to_snapshot(false);
		insta_settings.set_sort_maps(true);
		insta_settings.set_input_file(path);
		insta_settings.set_snapshot_suffix(path.file_stem().unwrap().to_string_lossy());

		run_lune_test(&input, true, insta_settings)
	});
}
