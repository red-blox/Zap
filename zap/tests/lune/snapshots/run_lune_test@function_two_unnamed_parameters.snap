---
source: zap/tests/lune/mod.rs
expression: result
input_file: zap/tests/files/function_two_unnamed_parameters.zap
---
Ok(
    0,
)
